src/pc/voice_chat.c

```
@@ -0,0 +1,438 @@
+    }
+    buffer->size -= bytes;
+    return bytes;
+}
+
+static void buffer_grow(struct VoiceBuffer* buffer, u32 bytes) {
+    if (buffer->dynamic && buffer->size + bytes > buffer->capacity) {
+        buffer->capacity = buffer->size + bytes;
+        if (buffer->capacity % 1024) buffer->capacity += 1024 - (buffer->capacity % 1024);
+        if (buffer->bytes) {
+            u8* new_bytes = malloc(buffer->capacity);
+            memcpy(new_bytes, buffer->bytes + buffer->tail, (buffer->size - buffer->tail));
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)